### ISSUE TITLE
Efficient way to always get n most recent partial results?

### ISSUE THREAD

**Issue by dangtony98** [react-native-voice/voice#369]
For context, I'm working on a voice recognition application that wishes to consider the last n words that are uttered by the user; the application doesn't care about any words that precede the last n-word window.

partialResults seems to return results from the beginning of start recognizing and clears when I either stop or cancel recognizing. Given this output, I'm currently slicing the last n words off the partialResults to be considered by my application but this feels inefficient in the order of O(n) ... 

Is there a more efficient way to limit partialResults to by default only return the n most recent results at all times?

**Comment by hongzhou85** [comment]
I'm also looking at this problem. it seems that this library does not provide a continual mode to get partial results.

I'm going to attempt detecting speech, then set a window of say 5 seconds to get partial results and repeat.

**Comment by tslater** [comment]
Yeah I just want a clear/easy way to reset partialResults to an empty string. I'm trying to cancel and/or stop and/or destroy...but i run into all sorts of errors and issues.

**Comment by chhornponleu** [comment]
it must be a bug. partial results should not always the same as results.

**Comment by aliyaus** [comment]
any update on this issue?

**Comment by safaiyeh** [comment]
This project has been deprecated and archived. Please migrate to [`expo-speech-recognition`](https://github.com/jamsch/expo-speech-recognition).
Thank you for your contributions! ❤️
